test_data/makedata_suites/560_smartgraph_edge_validator.js

```diff
@@ -0,0 +1,59 @@
+/* global fs, PWD, writeGraphData, getShardCount, getReplicationFactor,  print, progress, db, createSafe, _, semver */
+
+(function () {
+  return {
+    isSupported: function (currentVersion, oldVersion, options, enterprise, cluster) {
+      let current = semver.parse(semver.coerce(currentVersion));
+
+      return semver.gte(current, "3.9.0") && cluster && !options.readOnly;
+    },
+    checkData: function (options, isCluster, isEnterprise, dbCount, loopCount, readOnly) {
+      // depends on the 550_enterprise_graph.js to be there
+      print(`checking data ${dbCount} ${loopCount}`);
+      const ArangoError = require('@arangodb').ArangoError;
+
+      try {
+        const vColName = `patents_smart_${loopCount}`;
+        const eColName = `citations_smart_${loopCount}`;
+        const gName = `G_smart_${loopCount}`;
+        const remoteDocument = {
+          _key: "abc:123:def",
+          _from: `${vColName}/abc:123`,
+          _to: `${vColName}/def:123`
+        };
+        const localDocument = {
+          _key: "abc:123:abc",
+          _from: `${vColName}/abc:123`,
+          _to: `${vColName}/abc:123`
+        };
+        const testValidator = (colName, doc) => {
+          let col = db._collection(colName);
+          if (!col) {
+            throw new Error(`The smartGraph "${gName}" was not created correctly, collection ${colName} missing`);
+          }
+          try {
+            col.save(doc);
+            throw new Error(`Validator did not trigger on collection ${colName} stored illegal document`);
+          } catch (e) {
+            // We only allow the following two errors, all others should be reported.
+            if (e instanceof ArangoError) {
+              if (e.errorNum !== 1466 &&
```

Comment:
Not sure if the following comment applies here, but we might need to add a check for error code 4010 as well.

https://arangodb.slack.com/archives/C66C6TTV0/p1668431936231369?thread_ts=1668182220.607539&cid=C66C6TTV0